tests/test_plugin.py

```diff
@@ -465,6 +465,8 @@ def test_plugin_connected_hook_chaining(node_factory):
     ])
     assert len(l1.rpc.listpeers(l2id)['peers']) == 1
 
+    # Make sure we're note still disconnecting at this point!
+    wait_for(lambda: l3.rpc.listpeers(l1.info['id'])['peers'] == [])
```

Comment:
Mh, I think there is some similar flacky (maybe due to some copy-and-paste code) in https://github.com/ElementsProject/lightning/pull/6315#pullrequestreview-1492878509

The tests should be `test_restorefrompeer`